android/src/main/java/com/guichaguri/trackplayer/service/MusicService.java

```diff
@@ -127,6 +127,8 @@ public void onTaskRemoved(Intent rootIntent) {
         super.onTaskRemoved(rootIntent);
 
         if (manager == null || manager.shouldStopWithApp()) {
+            manager.getPlayback().stop();
+            destroy();
```

Comment:
I don't see how this fixes the problem. `stopSelf()` should automatically call `onDestroy`, which will cleanup the service resources.

Reply:
I investigated but onDestory was not called. I don't know where this behavior is documented.

related?: https://stackoverflow.com/questions/7236782/is-an-android-service-guaranteed-to-call-ondestroy/7237522#7237522

Reply:
I'm sorry. It was a misunderstanding.
OnDestroy is called, but I don't know if it's the effect of the release order, but the app state remains in memory.

Reply:
do we need to check if manager == null  before callin manager.getPlayback().stop();?

Reply:
You are right. Thanks.